File: python/get_commits.py
import requests

def get_commits(github_base_url :str, user_name :str, repository :dict, headers :dict) -> tuple[dict, str]:
    github_commits_url = f"{github_base_url}/repos/{user_name}/{repository['name']}/commits"
    response = requests.request("GET", github_commits_url, headers=headers)

    if response.status_code == 200:
        commits = response.json()
        return commits, ""
    else:
        return {}, f"Failed to retrieve commits. Status code: {response.status_code}"

File: go/get_commits.go
package main

import (
	"encoding/json"
	"fmt"
	"io"
	"log"
	"net/http"
)

func get_commits(client *http.Client, Bearer string, repository Repository) []Commit {

	fetch_commits_url := fmt.Sprintf("%s/repos/%s/%s/commits", GITHUB_BASE_URL, USER_NAME, repository.Name)
	req, err := http.NewRequest("GET", fetch_commits_url, nil)
	if err != nil {
		log.Fatalln(err)
	}
	req.Header.Set("Authorization", Bearer)

	resp, err := client.Do(req)
	if err != nil {
		log.Println("Error on response.\n[ERROR] -", err)
	}
	defer resp.Body.Close()

	body, err := io.ReadAll(resp.Body)
	if err != nil {
		log.Println("Error while reading the response bytes:", err)
	}

	var commits []Commit

	err = json.Unmarshal(body, &commits)
	if err != nil {
		log.Println("Error while marshalling the response bytes:", err)
	}

	return commits
}
